swap/src/monero.rs

```diff
@@ -99,6 +99,20 @@ impl Amount {
         self.0
     }
 
+    pub fn max_bitcoin_for_price(&self, ask_price: bitcoin::Amount) -> Result<bitcoin::Amount> {
+        let piconero_minus_fee = self.as_piconero().saturating_sub(MONERO_FEE.as_piconero());
+
+        if piconero_minus_fee == 0 {
+            bail!("The current Monero balance is smaller than the basic locking fee");
```

Comment:
I don't know if it's really the best idea to drop the connection when the balance is too low. It could lead to unexpected behavior and confuse the user of swap-cli, since he won't get any real feedback, but only some cryptic timeout error.

I still think returning a quote where the `max_quantity` is zero is the best workaround.

What do you think? @thomaseizinger

Reply:
Thanks for bringing this to my attention!

Yes, I think for a start, this function knows too much about the context it is called in. We should just return 0 here if that is the result of the computation.

If we still want to fail the quote protocol then we should do that one layer up!

Can we have a test run with both implementations and compare the user output?

Reply:
I just uploaded the suggestion from @binarybaron . But I think this should be a temporary solution because we are communicating to the user there is min_ buy amount and that could bring confusion. 

If there is not enough Monero to trade maybe we could give the SWAP user a graceful disconnect with a message or something like that